bin/wasm-node/rust/src/lib.rs

```diff
@@ -832,11 +863,36 @@ async fn handle_rpc(rpc: &str, client: &mut Client) -> (String, Option<String>)
     }
 }
 
+async fn announce_transaction(
```

Comment:
The logic of this thing should ideally be done in `NetworkService::announce_transaction`